rdtools/test/filtering_test.py

```diff
@@ -1,96 +1,123 @@
 
-        # Expected high and low tcell cutoffs to be non-inclusive.
-        expected_result = np.array([False, True, True, True, False])
-        self.assertListEqual(filtered.tolist(), expected_result.tolist())
+    # Expected high and low tcell cutoffs to be non-inclusive.
+    expected_result = np.array([False, True, True, True, False])
+    assert filtered.tolist() == expected_result.tolist()
 
 
-class ClipFilterTestCase(unittest.TestCase):
+def test_clip_filter():
     ''' Unit tests for inverter clipping filter.'''
 
-    def setUp(self):
-        self.power = pd.Series(np.arange(1, 101))
-        # Note: Power is expected to be Series object because clip_filter makes
-        #       use of the Series.quantile() method.
-
-    def test_clip_filter_upper(self):
-        filtered = clip_filter(self.power, quantile=0.98)
-
-        # Expect 99% of the 98th quantile to be filtered
-        expected_result = self.power < (98 * 0.99)
-        self.assertTrue((expected_result == filtered).all())
+    power = pd.Series(np.arange(1, 101))
+    # Note: Power is expected to be Series object because clip_filter makes
+    #       use of the Series.quantile() method.
+    filtered = clip_filter(power, quantile=0.98)
+    # Expect 99% of the 98th quantile to be filtered
+    expected_result = power < (98 * 0.99)
+    assert ((expected_result == filtered).all())
+
+
+@pytest.fixture
+def generate_power_time_series():
+    power_no_datetime_index = pd.Series(np.arange(1, 101))
+    power_datetime_index = pd.Series(np.arange(1, 101))
+    # Add datetime index to second series
+    time_range = pd.date_range('2016-12-02T11:00:00.000Z',
+                               '2017-06-06T07:00:00.000Z', freq='H')
+    power_datetime_index.index = pd.to_datetime(time_range[:100])
+    # Note: Power is expected to be Series object with a datetime index.
+    return power_no_datetime_index, power_datetime_index
+
+
+def test_logic_clip_filter(generate_power_time_series):
+    ''' Unit tests for geometric clipping filter.'''
+    power_no_datetime_index, power_datetime_index = \
+        generate_power_time_series
+    # Test that a Type Error is raised when a pandas series
+    # without a datetime index is used.
+    pytest.raises(TypeError,  logic_clip_filter,
+                  power_no_datetime_index)
+    # Expect none of the sequence to be clipped (as it's
+    # constantly increasing)
+    filtered, mask = logic_clip_filter(power_datetime_index)
+    assert (mask.all())
+
+
+def test_xgboost_clip_filter(generate_power_time_series):
+    ''' Unit tests for geometric clipping filter.'''
+    power_no_datetime_index, power_datetime_index = \
+        generate_power_time_series
+    # Test that a Type Error is raised when a pandas series
+    # without a datetime index is used.
+    pytest.raises(TypeError,  xgboost_clip_filter,
+                  power_no_datetime_index)
+    # Expect none of the sequence to be clipped (as it's
+    # constantly increasing)
```

Comment:
In addition to testing that the functions don't detect clipping when it isn't present, it might be good to test that they do detect clipping when it is 😄